cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -189,4 +189,75 @@ VAR_LEN_TYPES(IS_NOT_NULL, isnotnull)
 
 #undef IS_NOT_NULL
 
+FORCE_INLINE
+char* substr_utf8_int64_int64(int64 context, const char* input, int32 in_len,
+                              int64 offset64, int64 length, int32* out_len) {
+  if (length <= 0 || input == nullptr || in_len <= 0) {
+    *out_len = 0;
+    return nullptr;
+  }
+
+  int32 offset = static_cast<int32>(offset64);
+  int32 startIndex = offset - 1;  // offset is 1 for first char
+  if (offset < 0) {
```

Comment:
this doesn't sound correct - the if-else can be replaced with startIndex = offset

Reply:
offset starts from 1 in impala, sql Server and MySQL. adding same behaviour here